lib/util/dedent.py

```diff
@@ -0,0 +1,24 @@
+r"""
+This allows raw blocks like
+
+    variable = dedent(fr'''
+        some
+        content
+    ''')
+
+without the leading newline, any common leading whitespaces, and the trailing
+newline with 4 spaces. You might think using '''\ would eliminate the first
+newline, but the string is 'raw', it doesn't have escapes.
```

Comment:
The `4` refers specifically to the example given,
```diff
+    variable = dedent(fr'''
+        some
+        content
+    ''')
```
which uses 4 spaces to indent the last line.

The "trailing newline" mention is important - it's the main reason for this function.

```python
def a():
    variable = fr'''
        some
        content
    '''
    return variable
```
results in
```
>>> a()
'\n        some\n        content\n    '
```
quoting the comment:
- without the leading newline (`\n`)
- any common leading whitespaces (`        ` before `some` and `content`)
- and the trailing newline (`\n`)
- with 4 spaces (`    ` after the last `\n`)

If it was too confusing, I can probably reword it, or maybe add the one-line result to better explain what problem the function solves. Or maybe encase the block inside `def a()` to avoid confusion with module-level (zero indent) `variable = ` definition.

Reply:
> The "trailing newline" mention is important - it's the main reason for this function.

Yes, I probably wrongly worded my comment. The "trailing newline" is important to mention, I agree.

> The 4 refers specifically to the example given,

That is my misunderstanding. When I read those sentences, I thought they are for the function in general, not for the example. To clarify my thoughts, when I read it (I was reading docstring first, so at the moment I didn't know what the function code looks like) I thought:
> Why it removes ONLY 4 spaces? What if there is another indentation level? It will leave trailing spaces

So I checked the code and saw it removes all trailing spaces.